File: packages/native/ios/Nitro/OneNativeError.swift
import NitroModules

// nitro carries only a message across jsi, so a coded failure travels as
// "<code>: <message>" and the js entries split the stable code back out.
func oneNativeError(_ code: String, _ message: String) -> RuntimeError {
  return .error(withMessage: "\(code): \(message)")
}

File: packages/one/android/src/main/java/com/margelo/nitro/one/OneNativeError.kt
package com.margelo.nitro.one

// nitro hands js the printed stack trace of a rejected throwable as the error
// message. this one prints as exactly its message, with no class name or
// frames: "<code>: <message>" for a failure, whose stable code the js entries
// split back out, or the bare message for a call the platform cannot answer.
class OneNativeError(message: String) : Exception(message) {
    constructor(code: String, message: String) : this("$code: $message")

    override fun fillInStackTrace(): Throwable = this

    override fun toString(): String = message ?: ""
}
